### PR TITLE
logging: add ISO-like timestamp option - v1

### PR DESCRIPTION
**Pull request by jasonish**
Add a new log format specifier `%z` to use a more ISO-like timestamp format `YYYY-MM-DD HH:MM:SS` that is more commonly used in logs these days.

- Make this the default in the file log option.

Ticket: https://redmine.openinfosecfoundation.org/issues/5764

### CODE REVIEW

File: src/util-debug.c [hunk 1]
```diff
@@ -383,16 +383,33 @@ static SCError SCLogMessageGetBuffer(struct timeval *tval, int color, SCLogOPTyp
         strlcat(local_format, "%M", sizeof(local_format));
     char *temp_fmt = local_format;
     char *substr = temp_fmt;
+    struct tm local_tm;
 
-	while ( (temp_fmt = strchr(temp_fmt, SC_LOG_FMT_PREFIX)) ) {
+    while ((temp_fmt = strchr(temp_fmt, SC_LOG_FMT_PREFIX))) {
```
Comment (jasonish) [inline src/util-debug.c:388]: clang-format

File: src/util-debug.c [hunk 2]
```diff
@@ -588,7 +605,7 @@ static SCError SCLogMessageGetBuffer(struct timeval *tval, int color, SCLogOPTyp
             }
         }
         temp_fmt++;
-	}
+    }
```
Comment (jasonish) [inline src/util-debug.c:608]: clang-format

**Comment by suricata-qa** [comment]
Information:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 51883 | 0.01 | 

Pipeline 11284

File: doc/userguide/configuration/suricata-yaml.rst [hunk 3]
```diff
@@ -1850,8 +1854,8 @@ specified signs:
 
 ::
 
-  t:      Time, timestamp, time and date
-			example: 15/10/2010 - -11:40:07
+  z:      ISO-like formatted timestamp: YYYY-MM-DD HH:MM:SS
+  t:      Original Suricata log timestamp: DD/MM/YYYY -- HH:MM::SS
   p:      Process ID. Suricata's whole processing consists of multiple threads.
   i:      Thread ID. ID of individual threads.
   m:      Thread module name. (Outputs, Detect etc.)
```
Comment (jufajardini) [inline doc/userguide/configuration/suricata-yaml.rst:1861]: With the recent changes by Victor, I think that this section is missing a few options (D, M, S, T, from what I can see).

**Review by jufajardini — COMMENTED** [review]
Documentation-wise looks good - although it made me realize we are missing explaining a few of the format options - left a comment inline. Case this PR is ready to merge, I can submit one to fix that bit up.

**Comment by victorjulien** [comment]
Merged in #8360, thanks!

**Pull request closed without merging**